pkg/receive/handler.go

```diff
@@ -310,6 +310,12 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 		tenant = h.options.DefaultTenantID
 	}
 
+	// exit early if the request contained no data
+	if 0 == len(wreq.Timeseries) {
```

Comment:
Nit: just to make the intention of the comparison clearer we prefer to put the unknown on the left side
```suggestion
	if len(wreq.Timeseries) == 0 {
```